File: ADMIN/Example-API-Flutter/Example.UserService.API/Entities/ExamSet.cs
using Example.Common.Base;

namespace Example.UserService.API.Entities
{
    public class ExamSet : EntityAuditBase<long>
    {
        public string Name { get; set; }              // Tên bộ đề
        public string Description { get; set; }       // Mô tả
        public int RequiredTotalPoints { get; set; }  // Tổng điểm yêu cầu
        public int QuestionCount { get; set; }        // Số câu hỏi cần lấy
        public int EasyCount { get; set; }            // Số câu dễ
        public int MediumCount { get; set; }        // Số câu trung bình
        public int HardCount { get; set; }            // Số câu khó
        public int Type { get; set; }                 // 1=Thi thật, 2=Thi thử
        public int DurationMinutes { get; set; }      // Thời gian làm bài (phút)
        public int Status { get; set; }               // 1=Active, 2=Inactive
    }
}

File: lib/models/exam/exam_set.dart
/// Bộ đề thi — tương đương `ExamSetModel`.
class ExamSet {
  final int id;
  final String name;
  final String description;
  final int durationMinutes;
  final int questionCount;
  final int status;
  final int type;

  const ExamSet({
    required this.id,
    required this.name,
    required this.description,
    required this.durationMinutes,
    required this.questionCount,
    required this.status,
    required this.type,
  });

  factory ExamSet.fromJson(Map<String, dynamic> json) {
    return ExamSet(
      id: json['id'] as int? ?? 0,
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      durationMinutes: json['durationMinutes'] as int? ?? 60,
      questionCount: json['questionCount'] as int? ?? 0,
      status: json['status'] as int? ?? 0,
      type: json['type'] as int? ?? 0,
    );
  }
}
